### PR TITLE
chore(project): upgrade to typescript2

### LINKED ISSUE #2: Get rid of dependency to private typings repo
Currently we depend on `git@github.bus.zalan.do:graviton/graviton-typings.git`, we should change this...

### LINKED ISSUE #21: Bumb TypeScript Version to 2.0

### LINKED ISSUE #22: Upgrade nock
In projects having nock@8.x.x and this lib (which uses nock@7.x.x) as dependency a npm bug (https://github.com/npm/npm/issues/10727) causes problems when running `npm install` twice.

### PR DESCRIPTION
**Pull request by bzums**
* use typescript2
* upgrade npm dependencies
* get rid of global dependencies

close #2
close #21
close #22

### CODE REVIEW

**Comment by bzums** [comment]
👍

**Comment by bzums** [comment]
👍

**Comment by ISO50** [comment]
👍

File: package.json [hunk 1]
```diff
@@ -24,28 +24,36 @@
...
     "form-urlencoded": "^1.2.0",
     "http-status": "^0.2.0",
-    "nock": "^7.5.0",
+    "nock": "^8.0.0",
     "node-fetch": "^1.3.3",
     "node-uuid": "^1.4.7",
     "q": "^1.4.1"
   },
   "devDependencies": {
+    "@graviton/graviton-typings": "1.0.0",
```
Comment (bzums) [inline package.json]: @awaltert You are right, I created a ticket regarding that: https://github.com/zalando-incubator/lib-oauth-tooling/issues/25

**Pull request merged**